examples/geometry.shader_test

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[vertex shader]
+#version 430
+
+layout(location = 0) in vec4 position;
+layout(location = 1) in vec4 color_in;
+layout(location = 0) out vec4 color_out;
+
+void
+main()
+{
+        gl_Position = position;
+        color_out = color_in;
+}
+
+[geometry shader]
+#version 430
+
+layout (points) in;
+layout (triangle_strip, max_vertices = 6) out;
+
+layout(location = 0) in vec4 color_in[];
+
+layout(location = 0) out vec4 color_out;
+
+void main() {
+        gl_Position = gl_in[0].gl_Position;
+        color_out = color_in[0];
+        EmitVertex();
+
+        gl_Position = vec4(gl_in[0].gl_Position.x + 1.0,
+                           gl_in[0].gl_Position.y,
+                           0.0, 1.0);
+        color_out = color_in[0];
+        EmitVertex();
+
+        gl_Position = vec4(gl_in[0].gl_Position.x + 1.0,
+                           gl_in[0].gl_Position.y + 1.0,
+                           0.0, 1.0);
+        color_out = color_in[0];
+        EmitVertex();
+        EndPrimitive();
+
+
+        gl_Position = vec4(gl_in[0].gl_Position.x,
+                           gl_in[0].gl_Position.y + 1.0,
+                           0.0, 1.0);
+        color_out = color_in[0];
+        EmitVertex();
+
+        gl_Position = vec4(gl_in[0].gl_Position.x + 1.0,
+                           gl_in[0].gl_Position.y + 1.0,
+                           0.0, 1.0);
+        color_out = color_in[0];
+        EmitVertex();
+
+        gl_Position = gl_in[0].gl_Position;
+        color_out = color_in[0];
+        EmitVertex();
+        EndPrimitive();
+}
+
+[fragment shader]
+#version 430
+
+layout(location = 0) in vec4 color_in;
+layout(location = 0) out vec4 color_out;
+
+void
+main()
+{
+        color_out = color_in;
+}
+
+[vertex data]
+# position      color
+0/R32G32_SFLOAT 1/A8B8G8R8_UNORM_PACK32
+
+# a point with red color
+0 0          0xff0000ff
+
```

Comment:
Most of the Piglit tests draw a red colour for failure and green for success. It might be nice to keep up that tradition unless there is a good reason not to.

Reply:
Done.